app/Domain/Tokens/Token.h

```diff
@@ -0,0 +1,29 @@
+//
+// Created by Gabriel Noriega on 2018-02-15.
+//
+
+#ifndef SMALLWORLD_TOKEN_H
+#define SMALLWORLD_TOKEN_H
+
+#include "TokenType.h"
+#include "TokenSubtype.h"
+
+struct Token {
```

Comment:
Given our immutability-oriented programming, I suggest that m_inDecline and m_transient both be const bool, and that whenever the `goInDecline` and `setTransient` members are called, they return a new instance of the Token

Reply:
But what if a Race is not in decline anymore?

Reply:
Is it really necessary though?

Reply:
Well the whole architecture is based on the premise that every data types we have are immutable and return a copy of themselves upon mutation. This allows shallow equality checking by the store to update the views, rather than having to manually build an observer pattern on every data type